internal/app/secretless/handlers/pg/protocol/ssl.go

```diff
@@ -0,0 +1,149 @@
+package protocol
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net"
+)
+
+type options map[string]string
```

Comment:
This threw me a bit, because we _call_ it like this in backend.go:
```go
tlsConf, err := protocol.ResolveTLSConfig(h.BackendConfig.QueryStrings)
```
Yet here we have a special type, which is private, called `options`.  Ofc it works because they're both just `map[string]string`.  But I'd say that you _either_ want to just omit this type, and use the `map[string]string` everywhere, _or_ expose the `protocol.Options` type, and use that at the calling site as well.